troposphere/static/js/components/images/detail/availability/AvailabilityView.react.js

```diff
@@ -1,32 +1,56 @@
 import React from "react";
 import Backbone from "backbone";
 import stores from "stores";
-
+import Code from "components/common/ui/Code.react.js";
 
 export default React.createClass({
     displayName: "AvailabilityView",
 
     propTypes: {
-        version: React.PropTypes.instanceOf(Backbone.Model).isRequired
+        version: React.PropTypes
+            .instanceOf( Backbone.Model ).isRequired
     },
 
-    renderProviderMachine: function(provider_machine) {
+    renderProviderMachine( provider ) {
+        let { isSummary } = this.props;
+
         // Hide 'end-dated' provider_machines
-        let endDate = provider_machine.get("end_date");
+        let endDate = provider.get( "end_date" );
         if (endDate && endDate.isValid()) return;
 
+        // Assign strings for render
+        let machineID = provider.get( "uuid" );
+        let providerName = provider.get( "provider" ).name;
+
+        // Optionally render the UUID if not set as
+        // summary by parent
+        let optMachineID = isSummary ? null
```

Comment:
I'd like to see how you prefer to do this. The link above is to this line in the diff, but I'm assuming you intended to link to another solution? Could you add a snippet here? 

Otherwise I think you are comparing:
```
let dependentVar;
if (factorVar) {
    dependentVar = ( ... )
}

// vs

let dependentVar = factorVar ? ( ... ) : null;
```